resolvers/simple.resolver.js

```diff
@@ -22,16 +22,19 @@ module.exports = async (job, log, data) => {
 
   allScripts.forEach((script) => {
     let scriptContents = log.substr(log.indexOf(script));
-    scriptContents = scriptContents.split('\n').slice(1).join('\n');
-    scriptContents = scriptContents.substr(0, scriptContents.indexOf('" exited with ')).trim();
-    scriptContents = scriptContents.split('\n').slice(0, -1).join('\n');
-    scriptContents = scriptContents.trim();
+    const exitCode = /exited\swith\s(\d)/g.exec(scriptContents);
+    if (Number(exitCode[1]) !== 0) {
```

Comment:
What if by some kind of magic the log doesn't contain `exited with ...`? This line will fail, so I'd suggest making sure that `scriptContents` isn't falsey (something like: `exitCode && Number(exitCode[1]) !== 0)`)

It's highly important to do null-checks IMHO, since TravisBuddy is live in production and since we don't control the build log that TravisCI sends us, unexpected things could happen :)

Reply:
Thanks for pointing this out.